src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/tools/HerculesPickaxe.java

```diff
@@ -25,16 +27,36 @@ public HerculesPickaxe(ItemGroup itemGroup, SlimefunItemStack item, RecipeType r
     @Override
     public @Nonnull ToolUseHandler getItemHandler() {
         return (e, tool, fortune, drops) -> {
-            if (SlimefunTag.ORES.isTagged(e.getBlock().getType())) {
-                if (e.getBlock().getType() == Material.IRON_ORE) {
-                    drops.add(new CustomItemStack(SlimefunItems.IRON_DUST, 2));
-                } else if (e.getBlock().getType() == Material.GOLD_ORE) {
-                    drops.add(new CustomItemStack(SlimefunItems.GOLD_DUST, 2));
-                } else {
-                    for (ItemStack drop : e.getBlock().getDrops(tool)) {
-                        drops.add(new CustomItemStack(drop, drop.getAmount() * 2));
+            Material mat = e.getBlock().getType();
+
+            if (SlimefunTag.ORES.isTagged(mat)) {
+                if (Slimefun.getMinecraftVersion().isAtLeast(MinecraftVersion.MINECRAFT_1_17)) {
+                    switch(mat) {
+                        case DEEPSLATE_IRON_ORE:
+                            drops.add(new CustomItemStack(SlimefunItems.IRON_DUST, 2));
+                            break;
+                        case DEEPSLATE_GOLD_ORE:
+                            drops.add(new CustomItemStack(SlimefunItems.GOLD_DUST, 2));
+                            break;
+                        case COPPER_ORE:
+                        case DEEPSLATE_COPPER_ORE:
+                            drops.add(new CustomItemStack(SlimefunItems.COPPER_DUST, 2));
+                            break;
```

Comment:
Missing `default:` statement

Reply:
this section is about 1.17 ores, if the block type does not match here, it should then go to next switch

Reply:
This switch is still missing a `default: break;` statement though.
Having no 'default' statement usually implies that there are no unhandled possible values.
But this `switch` does not cover all values of the `Material` enum, hence why it definitely needs a `default` statement from a semantical view.